.github/workflows/reusable_basic.yml

```diff
@@ -181,11 +168,11 @@ jobs:
         ${{ matrix.compiler.cxx == 'icpx' && '. /opt/intel/oneapi/setvars.sh' || true }}
         LD_LIBRARY_PATH="${{env.BUILD_DIR}}/lib/:${LD_LIBRARY_PATH}" ctest --output-on-failure
 
-    - name: Check coverage
+    - name: Check os coverage
       if:  ${{ matrix.build_type == 'Debug' && matrix.compiler.c == 'gcc' }}
       working-directory: ${{env.BUILD_DIR}}
       run: |
-        export COVERAGE_FILE_NAME=${{env.COVERAGE_NAME}}-${{matrix.os}}-shared-${{matrix.shared_library}}-no_hwloc-${{matrix.disable_hwloc}}
+        export COVERAGE_FILE_NAME=${{env.COVERAGE_NAME}}-${{matrix.ubuntu_ver}}-shared-${{matrix.shared_library}}-no_hwloc-${{matrix.disable_hwloc}}
```

Comment:
ok, that's not good - the name of the artifact will be `cov-22.04-shared...` - we should definitely keep the old name with "ubuntu" in it - as we may have more OSes than just ubuntu - perhaps you should use the var as before `os` instead of `ubuntu_ver` -- so basically turn back what it was in matrix - `os: ['ubuntu-20.04', 'ubuntu-22.04']`

For unity, the same approach in FastBuilds - `os: ['ubuntu-20.04', 'ubuntu-22.04']` instead of `ubuntu_ver: <number>`